src/writer/app_runner.py

```
@@ -1245,6 +1261,52 @@ def reload_code_from_saved(self) -> None:
+            wait_timeout: Maximum seconds to wait for app to be ready. None = don't wait.
+        
+        Returns:
+            True if app started successfully, False otherwise.
+        """
+        if not self.is_app_process_server_ready.is_set():
+            return False
+        
+        try:
+            run_code = self.load_persisted_script()
+            if self.mode != "edit":
+                raise PermissionError("Cannot update code in non-edit mode.")
```

Ruff (0.14.5)

Abstract `raise` to an inner function

(TRY301)